test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -4210,6 +4211,42 @@ TEST_F(AtenXlaTensorTest, TestRandperm) {
                           cpp_test::GetIgnoredCounters());
 }
 
+TEST_F(AtenXlaTensorTest, TestRandpermOutWithoutGenerator) {
+  int n = 5;
+  torch::Tensor a = torch::randint(16, {n}, torch::TensorOptions(torch::kLong));
+  ForEachDevice([&](const torch::Device& device) {
+    torch::Tensor xla_a = CopyToDevice(a, device);
+    torch::Tensor xla_b = torch::randperm_out(xla_a, n);
+
+    torch::Tensor xla_b_cpu = CopyToDevice(xla_b, torch::kCPU);
+    std::vector<int64_t> shuffle_data(xla_b_cpu.data_ptr<int64_t>(),
+                                      xla_b_cpu.data_ptr<int64_t>() + n);
+    EXPECT_TRUE(shuffle_data.size() == n && xla::IsPermutation(shuffle_data));
+  });
+
+  ExpectCounterChanged("xla::randperm_out", cpp_test::GetIgnoredCounters());
+  ExpectCounterNotChanged("aten::.*", cpp_test::GetIgnoredCounters());
+}
+
+TEST_F(AtenXlaTensorTest, TestRandpermOutWithGenerator) {
```

Comment:
Generator should force us to fall back to CPU, so you don't need to compare the results. Instead, let's just make sure that xla counter is not changed, but aten counter is.

Having said that, this test case will be a lot shorter -- can you move/merge this and the previous into a single test case `AtenXlaTensorTest :: TestRandPermOut` ?